### PR TITLE
Add support for Ctrl+# keys

### LINKED ISSUE #3507: ctrl + # not being processed by Bash correctly
I'm sorry if this has been reported before, spent some time searching through the issues but couldn't find it.

Right now, if I type ctrl-NUM, it seems like it's not being picked up by the terminal app properly. Specifically, I've been trying to get the ctrl-6 vim behaviour to work, but I can't figure out if it's possible.

Thanks!

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

Fixes the <kbd>Ctrl+Num</kbd> keys in both conhost and the Terminal. These keys are supposed to be mapped to specific characters according to [this doc](https://vt100.net/docs/vt220-rm/table3-5.html). Now we actually handle them correctly.

## PR Checklist
* [x] Closes #3507 
* [x] I work here
* [x] Tests added/passed
* [n/a] Requires documentation to be updated

## Validation Steps Performed

* Ran test
* tested in `gnome-terminal` with `showkeys -a`
* tested in conhost with `showkeys -a`
* tested in Windows Terminal with `showkeys -a`

### CODE REVIEW

**Comment by DHowett-MSFT** [comment]
@msftbot merge this in 1 minute

**Comment by DHowett-MSFT** [comment]
(The test itself seems to be failing?)

File: src/terminal/input/terminalInput.cpp [hunk 1]
```diff
@@ -185,13 +185,31 @@ static constexpr std::array<TermKeyMap, 22> s_modifierKeyMapping{
...
-static constexpr std::array<TermKeyMap, 6> s_simpleModifiedKeyMapping{
+static constexpr std::array<TermKeyMap, 14> s_simpleModifiedKeyMapping{
     TermKeyMap{ VK_BACK, CTRL_PRESSED, L"\x8" },
     TermKeyMap{ VK_BACK, ALT_PRESSED, L"\x1b\x7f" },
     TermKeyMap{ VK_BACK, CTRL_PRESSED | ALT_PRESSED, L"\x1b\x8" },
     TermKeyMap{ VK_TAB, CTRL_PRESSED, L"\t" },
     TermKeyMap{ VK_TAB, SHIFT_PRESSED, L"\x1b[Z" },
     TermKeyMap{ VK_DIVIDE, CTRL_PRESSED, L"\x1F" },
+
+    // GH#3507 - We should also be encoding Ctrl+# according to the follwoing table:
```
Comment (DHowett-MSFT) [inline src/terminal/input/terminalInput.cpp]:
```suggestion
    // GH#3507 - We should also be encoding Ctrl+# according to the following table:
```

**Comment by zadjii-msft** [comment]
(I'm holding this till I merge #4988 and #4964)

**Comment by Zenexer** [comment]
@zadjii-msft Does this also send the same codes when <kbd>Shift</kbd> is held?  Some applications--e.g., mosh--advise users to press a combination such as <kbd>Ctrl</kbd> + <kbd>^</kbd> to send `^^`, which is often interpreted as requiring that <kbd>Shift</kbd> be held even though it's not necessary.  <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + `#` is bound by default in some terminals (including Windows Terminal), but some users may wish to unbind them, in which case they should send the correct codes.

**Pull request merged**